File: 1500-count-largest-group/count-largest-group.py
class Solution:
    def countLargestGroup(self, n: int) -> int:
        freq = [0] * 37

        for x in range(1, n + 1):
            s = 0
            while x:
                s += x % 10
                x //= 10
            freq[s] += 1

        mx = max(freq)
        return freq.count(mx)

File: 1500-count-largest-group/count-largest-group.java
class Solution {
    public int countLargestGroup(int n) {
        int[] freq = new int[37];

        for (int x = 1; x <= n; x++) {
            int sum = 0;
            int num = x;

            while (num > 0) {
                sum += num % 10;
                num /= 10;
            }

            freq[sum]++;
        }

        int max = 0;
        for (int x : freq) {
            max = Math.max(max, x);
        }

        int ans = 0;
        for (int x : freq) {
            if (x == max) {
                ans++;
            }
        }

        return ans;
    }
}